src/cloud/social/alias/randombytes.js

```diff
@@ -1,33 +1,55 @@
 // Alternative to crypto-browserify's randombytes shim which
-// falls back to Math.random when crypto.getRandomValues is
+// falls back to freedom.js core.crypto when crypto.getRandomValues is
 // unavailable, which is currently the case in Firefox web workers.
-// TODO: Fallback to something better such as freedomjs' core.crypto
-//       module (difficult because it is an asynchronous API).
+// TODO: post FF48 (2016-08-02) switch to just native crypto.getRandomValues
 
-// Firefox raises an error if we so much as even try
-// to *access* crypto, hence this bizarre construction.
-var cryptoAvailable = true;
-try {
-  if (crypto) {}
-} catch(e) {
-  console.warn('crypto unavailable, falling back to Math.random');
-  cryptoAvailable = false;
+var nativeCrypto = typeof crypto !== 'undefined';
+var refreshBuffer, bufferRemaining;
```

Comment:
Maintaining two variables which are essentially the inverse of each other - `bufferRemaining` and `offset` is a code smell.

I think you get away with two globals: `buffer` and `offset`.